internal/authentication/tokengetter.go

```diff
@@ -19,6 +21,15 @@ type TokenGetter struct {
 	mu                 sync.RWMutex
 }
 
+type ServiceAccountNotFoundError struct {
+	ServiceAccountName string // The name of the missing ServiceAccount.
+}
+
+// Error implements the error interface for ServiceAccountNotFoundError.
+func (e *ServiceAccountNotFoundError) Error() string {
+	return fmt.Sprintf("ServiceAccount \"%s\" not found: Unable to authenticate with the Kubernetes cluster.", e.ServiceAccountName)
```

Comment:
Some style suggestions to make this error message follow standard conventions:
```suggestion
	return fmt.Sprintf("service account %q not found: unable to authenticate with the Kubernetes cluster", e.ServiceAccountName)
```

Reply:
Also, should we include the expected namespace in the error message as well?

Reply:
added the same